### PR TITLE
Annotation filters UI

### PR DESCRIPTION
**Pull request by vnishukov**
### Motivation and context
[1418: CVAT new UI: find the way to simplify the syntax for filters](https://github.com/openvinotoolkit/cvat/issues/1418)

### How has this been tested?

- Go to annotation page
- Click to filters input
- Play around with form
- **Shift+Left click: add open bracket for item**
- **Shift+Right click: add close bracket for item**
- **Alt+Left click: remove open bracket for item**
- **Alt+Right click: remove close bracket for item**
- **Right click on filter input open confirm dialog to clean up filters**

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat-ui/src/components/annotation-page/annotation-filter/annotation-filter-pane/annotation-filter-pane.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    };
+
+    const isGroupingValid = (): boolean => {
+        let openBraces = 0;
+        let closeBraces = 0;
+        filters.forEach((filter: any) => {
+            openBraces += filter.left.length;
+            closeBraces += filter.right.length;
+        });
+        if (openBraces === 0 && closeBraces === 0) return true;
```
Comment (vnishukov) [inline cvat-ui/src/components/annotation-page/annotation-filter/annotation-filter-pane/annotation-filter-pane.tsx:68]: Remains from old realization. Need to get rid of it

**Comment by vnishukov** [comment]
Opened a new one

**Comment by nmanovic** [comment]
@vnishukov, What is the reason to close it?

**Comment by vnishukov** [comment]
New one was created to keep commit tree clean. Please see https://github.com/openvinotoolkit/cvat/pull/2743

**Pull request closed without merging**